wordpress-mobile/WordPress-iOS
# Domains: Added loading spinner to Choose Domain button

Fixes #17195. This PR adds a loading spinner and disables the Choose Domains button while we prepare to show the webview in the domains flow.

![domains-2](https://user-images.githubusercontent.com/4780/135673292-69404fc8-3144-4249-a124-3f3bb702264b.gif)

### To Test

I'm not as familiar with you both about the possible domains flow and I still can't complete the flow properly due to the dismissing webview issue (I'll try resetting everything soon to see if I can get past that), so hopefully you can double check I've covered all the situations here.

* Open a site with a premium plan and go to Domains
* Tap Search for a domain
* Find a domain and tap Choose Domain.
* You should see the button change to a spinner while the webview is presented. If you dismiss the webview the button should return to normal.

Please also see the associated [WPAuthenticator PR](https://github.com/wordpress-mobile/WordPressAuthenticator-iOS/pull/619).

## Regression Notes

1. Potential unintended areas of impact

None

2. What I did to test those areas of impact (or what existing automated tests I relied on)
3. What automated tests I added (or what prevented me from doing so)

N/A

PR submission checklist:
